infer/lib/python/inferlib/capture/gradle.py

```diff
@@ -31,9 +29,92 @@ def gen_instance(*args):
 create_argparser = util.base_argparser(MODULE_DESCRIPTION, MODULE_NAME)
 
 
+def extract_filepath(parts):
+    if not parts:
+        return ([], None)
+    path = ' '.join(parts)
+    if os.path.isfile(path):
+        return ([], path)
+    remainder, path = extract_filepath(parts[1:])
+    return ([parts[0]] + remainder, path)
+
+
+# Please run the doctests using:
+# $ python -m doctest -v gradle.py
+def extract_all(javac_arguments):
+    """Extract Java filenames and Javac options from the Javac arguments.
+
+    >>> os.path.isfile = lambda s: s[1:].startswith('path/to/')
+    >>> extract_all([])
+    {'files': [], 'opts': []}
+    >>> extract_all(['-opt1', 'optval1', '/path/to/1.java'])
+    {'files': ['/path/to/1.java'], 'opts': ['-opt1', 'optval1']}
+    >>> extract_all(['-opt1', 'optval1', '/path/to/a', 'b/1.java'])
+    {'files': ['/path/to/a b/1.java'], 'opts': ['-opt1', 'optval1']}
+    >>> extract_all(['-opt1', 'opt', 'val1', '/path/to/1.java'])
+    {'files': ['/path/to/1.java'], 'opts': ['-opt1', 'opt val1']}
+    >>> extract_all(['-opt1', '/path/to/a', 'b/c', 'd/1.java', '-opt2'])
+    {'files': ['/path/to/a b/c d/1.java'], 'opts': ['-opt1', '-opt2']}
+    >>> extract_all(['-opt1', 'optval1', '-path/to/1.java'])
+    {'files': ['-path/to/1.java'], 'opts': ['-opt1', 'optval1']}
+    >>> extract_all(['-opt1', 'optval1', '/path/to/', '-1.java'])
+    {'files': ['/path/to/ -1.java'], 'opts': ['-opt1', 'optval1']}
+    >>> extract_all(['undef1', 'undef2'])
+    {'files': [], 'opts': ['undef1 undef2']}
```

Comment:
To be honest this can only be a best-effort attempt and it's hard to know which one will be best in practice so I'm fine with whichever. I wouldn't spend too much time on this. As you mentioned, time would be better spent writing a better integration from scratch using a plugin, or find a way to extract some sort of compilation database from Gradle and create a build-system-neutral interface in infer to read these that could be re-used for other build systems for which infer doesn't have a good integration.